packages/store/addon/-private/ts-interfaces/minimum-adapter-interface.ts

```diff
@@ -0,0 +1,500 @@
+  - issue the request to the adapter
+  - await the adapter's response
+    - if an error occurs reject with the error
+    - if no error
+      - if there is response data
+        - pass the response data to the appropriate serializer
+        - update the cache using the JSON:API formatted data from the serializer's response
+      - return the primary record(s) associated with the request
+
+
+  ### Implementing an Adapter
+
+  There are seven required adapter methods, one for each of
+  the primary request types that EmberData issues.
+
+  They are:
+
+  - findRecord
+  - findAll
+  - queryRecord
+  - query
+  - createRecord
+  - updateRecord
+  - deleteRecord
+
+  Each of these request types has a matching store method that triggers it
+  and matching `requestType` that is passed to the serializer's
+  `normalizeResponse` method.
+
+  If your app only reads data but never writes data, it is not necessary
+  to implement the methods for create, update, and delete.
+
+
+  data
+  normalizing data from the server API format into JSON:API, and
+  another for serializing records via `Snapshots` into the expected
+  server API format.
+
+  To implement a serializer, export a class that conforms to the structure
+  described by the [MinimumSerializerInterface](/ember-data/release/classes/MinimumSerializerInterface)
+  from the `app/serializers/` directory. An example is below.
+
+  ```ts
+  import EmberObject from '@ember/object';
+
+  async function fetchData(url, options = {}) {
+    let response = await fetch(`./${modelName}s/${id}`, options);
+    return response.toJSON();
+  }
+
+  export default class ApplicationAdapter extends EmberObject {
+    findRecord(_, { modelName }, id) {
+      return fetchData(`./${modelName}s/${id}`);
+    }
+  }
+  ```
+
+  ### Adapter Resolution
+
+  `store.adapterFor(name)` will lookup adapters defined in
+  `app/adapters/` and return an instance. If no adapter is found, an
+  error will be thrown.
+
+  `adapterFor` first attempts to find an adapter with an exact match on `name`,
+  then falls back to checking for the presence of an adapter named `application`.
+
+  ```ts
+  store.adapterFor('author');
+
+  // lookup paths (in order) =>
+  //   app/adapters/author.js
+  //   app/adapters/application.js
+  ```
+
+  Most requests in EmberData are made with respect to a particular `type` (or `modelName`)
+  (e.g., "get me the full collection of **books**" or "get me the **employee** whose id is 37"). We
+  refer to this as the **primary** resource `type`.
+
+  Typically `adapterFor` will be used to find an adapter with a name matching that of the primary
+  resource `type` for the request, falling back to the `application` adapter for those types that
+  do not have a defined adapter. This is often described as a `per-model` or `per-type` strategy
+  for defining adapters. However, because APIs rarely define endpoints per-type but rather
+  per-API-version, this may not be a desired strategy.
```

Comment:
I think it seems a bit off to say that this may not be a desired strategy but not explain an alternative path?

Reply:
@igorT the thing to do (forwarding adapter) is a little hard to show succinctly and I don't know how we feel linking to a twiddle, maybe just link a gist?